Gems/Atom/RPI/Code/Tests/Material/MaterialSourceDataTests.cpp

```diff
@@ -465,6 +446,88 @@ namespace UnitTest
+            {
+                "materialType": "test.materialtype", // Doesn't matter, this isn't loaded
+                "properties": {
+                    "groupA": {
+                        "myBool": true,
+                        "myInt": 5,
+                        "myFloat": 0.5
+                    },
+                    "groupB": {
+                        "myFloat2": [0.1, 0.2],
+                        "myFloat3": [0.3, 0.4, 0.5],
+                        "myFloat4": [0.6, 0.7, 0.8, 0.9],
+                        "myString": "Hello"
+                    }
+                }
+            }
+        )";
+
+        MaterialSourceData material;
+        LoadTestDataFromJson(material, inputJson);
+        material.ConvertToNewDataFormat();
+
+        MaterialSourceData expectedMaterial;
+        expectedMaterial.m_materialType = "test.materialtype";
+        expectedMaterial.SetPropertyValue(Name{"groupA.myBool"}, true);
+        expectedMaterial.SetPropertyValue(Name{"groupA.myInt"}, 5);
+        expectedMaterial.SetPropertyValue(Name{"groupA.myFloat"}, 0.5f);
+        expectedMaterial.SetPropertyValue(Name{"groupB.myFloat2"}, Vector2(0.1f, 0.2f));
+        expectedMaterial.SetPropertyValue(Name{"groupB.myFloat3"}, Vector3(0.3f, 0.4f, 0.5f));
+        expectedMaterial.SetPropertyValue(Name{"groupB.myFloat4"}, Vector4(0.6f, 0.7f, 0.8f, 0.9f));
+        expectedMaterial.SetPropertyValue(Name{"groupB.myString"}, AZStd::string{"Hello"});
+
+        CheckEqual(expectedMaterial, material);
+    }
+    
+    TEST_F(MaterialSourceDataTests, TestPropertyValues)
+    {
+        MaterialSourceData material;
+        
+        Name foo{"foo"};
+        Name bar{"bar"};
+        Name baz{"baz"};
+
+        EXPECT_EQ(0, material.GetPropertyValues().size());
+        EXPECT_FALSE(material.HasPropertyValue(foo));
+        EXPECT_FALSE(material.HasPropertyValue(bar));
+        EXPECT_FALSE(material.HasPropertyValue(baz));
+        EXPECT_FALSE(material.GetPropertyValue(foo).IsValid());
+        EXPECT_FALSE(material.GetPropertyValue(bar).IsValid());
+        EXPECT_FALSE(material.GetPropertyValue(baz).IsValid());
+
+        material.SetPropertyValue(Name{"foo"}, 2);
+        material.SetPropertyValue(Name{"bar"}, true);
+        material.SetPropertyValue(Name{"baz"}, 0.5f);
+        
+        EXPECT_EQ(3, material.GetPropertyValues().size());
+        EXPECT_TRUE(material.HasPropertyValue(foo));
+        EXPECT_TRUE(material.HasPropertyValue(bar));
+        EXPECT_TRUE(material.HasPropertyValue(baz));
+        EXPECT_TRUE(material.GetPropertyValue(foo).IsValid());
+        EXPECT_TRUE(material.GetPropertyValue(bar).IsValid());
+        EXPECT_TRUE(material.GetPropertyValue(baz).IsValid());
+        EXPECT_EQ(material.GetPropertyValue(foo).GetValue<int32_t>(), 2);
+        EXPECT_EQ(material.GetPropertyValue(bar).GetValue<bool>(), true);
+        EXPECT_EQ(material.GetPropertyValue(baz).GetValue<float>(), 0.5f);
```

Comment:
I might be wrong, but I think that for floats you've got to use `EXPECT_FLOAT_EQ`, or you can use `EXPECT_NEAR` and specify a tolerance.

Reply:
The test passes so it should be fine :)
I think the reason it's fine is because no calculations are being done here, the values are just copied in and out, so any imprecision is identical between SetPropertyValue, GetValue, and EXPECT_EQ.

Reply:
Please check that `EXPECT_EQ` is not converting them to integers and therefore 0.5f becomes 0 and comparing `0 == 0`.